### PR TITLE
View entry

### PR DESCRIPTION
**Pull request by graceolivia**
Add links on the index page to allow the user to click on each title and go to a page with a single entry. 

Added:

- One function `get_journal_entry!` to get a single journal entry by id in EChronicler.Models.JournalEntry
- a `show` function in the Journal Entry Controller to call one page
- a new line in the router
- links to each individual entry page on the Titles on the main page
- HTML file for individual pages

Tests:
- One test for `JournalEntry.get_journal_entry!` in `test/e_chronicler/models/journal_entry_test.exs`
- Two tests for individual journal entry pages in `test/e_chronicler_web/controllers/journal_entry_controller_test.exs`

### CODE REVIEW

File: lib/e_chronicler/models/journal_entry.ex [hunk 1]
```diff
@@ -42,4 +42,6 @@ defmodule EChronicler.Models.JournalEntry do
     |> Repo.insert()
   end
 
+  def get_journal_entry!(id), do: Repo.get!(JournalEntry, id)
```
Comment (hnlee) [inline lib/e_chronicler/models/journal_entry.ex]: Should this function really fail passively when there's no entry with the given id? I'll note that you don't have a test for this case. You probably want to include that test case in both your model and controller tests.

What do you want to to tell the controller if there is no journal entry? And what should the controller do?

File: test/e_chronicler_web/controllers/journal_entry_controller_test.exs [hunk 2]
```diff
@@ -17,5 +17,12 @@ defmodule EChroniclerWeb.JournalEntryControllerTest do
     assert html_response(conn, 200) =~ "Supercilious"
   end
 
+  test "Single Journal Entry Is Displayed At /:id", %{conn: conn} do
+    {:ok, journal_entry} = EChronicler.Models.JournalEntry.create_journal_entry(%{author: "Bob", title: "Hello", entry: "Test."})
+    {:ok, _journal_entry_irrelevant} = EChronicler.Models.JournalEntry.create_journal_entry(%{author: "Betty", title: "Goodbye", entry: "Test."})
+    conn = get(conn, "/#{journal_entry.id}")
+    assert html_response(conn, 200) =~ "Hello"
+    assert !String.contains?(html_response(conn, 200), "Goodbye")
```
Comment (hnlee) [inline test/e_chronicler_web/controllers/journal_entry_controller_test.exs]: Your tests will probably be less brittle if you assert directly on the properties of the structs, e.g. 

```elixir
assert html_response(conn, 200) =~ journal_entry.title
refute html_response(conn, 200) =~ other_journal_entry.title
```

**Review by hnlee — CHANGES_REQUESTED** [review]

File: test/e_chronicler_web/controllers/journal_entry_controller_test.exs (hunk 2 above)
Comment (breadoliveoilsalt) [reply test/e_chronicler_web/controllers/journal_entry_controller_test.exs]: Approved from me so long as Hana's comments are met!  I steered Grace towards the `String.contains`...and I'm very glad now to learn about `refute`!

Comment (graceolivia) [reply test/e_chronicler_web/controllers/journal_entry_controller_test.exs]: Thank you! I'm also glad to learn about `refute`

File: lib/e_chronicler/models/journal_entry.ex (hunk 1 above)
Comment (graceolivia) [reply lib/e_chronicler/models/journal_entry.ex]: I think it should display a particular error page, like a 404 that says "entry not found." `Ecto.get!` throws an error and `Ecto.get` returns `nil` if nothing is found, not sure which one makes more sense to use

**Review by breadoliveoilsalt — APPROVED** [review]

**Review by hnlee — APPROVED** [review]

**Pull request merged**